### ISSUE TITLE
Google Maps API key needed

### ISSUE THREAD

**Issue by answerquest** [OneBusAway/onebusaway-gtfs-realtime-visualizer#11]
Their terms of use have changed recently, and I guess we need to put in our API key somewhere. Pls suggest where.

Screenshot of page: 
![click here](https://i.imgur.com/RiiOHtT.jpg)  

[Link that google takes us to](https://developers.google.com/maps/documentation/javascript/error-messages?utm_source=maps_js&utm_medium=degraded&utm_campaign=keyless#api-key-and-billing-errors).

**Comment by barbeau** [comment]
@answerquest To be honest I'm not sure the best way to configure this, as I didn't create this project.  Ideally we'd provide it as a command-line parameter if we want to allow it to be configured by the user at runtime, but I'm not sure the best way to accomplish this given how the project is set up.  Alternately we could bake in a development key.

**Comment by answerquest** [comment]
@barbeau it's not too bad actually as I wanted the screen to be darkened anyways ;)

So yes, my thought was to look for the specific code in the source and.. Hey I found it! Last commit on one of the files was: "[Remove Google Maps API key.](https://github.com/OneBusAway/onebusaway-gtfs-realtime-visualizer/commit/1de077a55690d8e8b87eb6da1f059b70158e2b73#diff-c3459aa55bf6d68b02b0e55ddf64a31e)". :1234:  :dancing_men: :rofl: 
So I'll just insert my key in here at this point and give it a spin.

**Comment by barbeau** [comment]
Good job tracking the commit down!  Yes, just editing the code and adding your own Maps API key there is the simplest solution to get it working - this is the "bake in a development key" approach.  But this doesn't help someone if they are trying to run the JAR without recompiling the project themselves.  And that's where I'm not immediately clear as to the best way of providing the key.

**Comment by answerquest** [comment]
Update: working well with the baked-in key.
[![screenshot](https://i.imgur.com/RbLCSJwm.jpg)](https://i.imgur.com/RbLCSJw.jpg)

**Comment by barbeau** [comment]
@answerquest Is this using your own Google Maps API key or the one that was originally included in this project in the commit you referenced?  If it's the one that was included in the project we can just revert that previous commit and allow using it for demos.

**Comment by answerquest** [comment]
@barbeau no, I'm using my own API key, not the original one. I'll recommend expiring old keys (i.e. from the google api a/c settings) as with the tighter limits coming these will be vulnerable to be scraped and used by unknown folks or bots, and there's a risk of incurring bills unexpectedly.  

The tool ***works*** without the API key also, the map displays, albeit in a darkened way as shown above, but still, functional enough. It's not a breaking issue. 

IMHO It'll be better to work on parameterizing the key and making it part of the command line, or having a config file where user can save in their key. And I think there may be more options people might want to add if there's a config file, like including more fields in the popup which are feed-specific. 

Another track: the mapping itself could be moved to leaflet / mapbox / openlayers.. but that would involve major re-writes in the way points and lines are drawn on it. See what works better.

I'll close this issue for now since I have a workaround fix for my needs, pls feel free to re-open or create new when development for a permanent fix takes place.

**Comment by barbeau** [comment]
@answerquest thanks, I agree that we need to have a configurable API key. Let's leave this open so it's obvious for others that this is a known issue.
